### PR TITLE
solaar: init at 0.9.2

### PR DESCRIPTION
**Pull request by spinus**

### CODE REVIEW

File: pkgs/applications/misc/solaar/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  postInstall = ''
+    wrapProgram "$out/bin/solaar" \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      --prefix GI_TYPELIB_PATH : "$GI_TYPELIB_PATH"
+    wrapProgram "$out/bin/solaar-cli" \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      --prefix GI_TYPELIB_PATH : "$GI_TYPELIB_PATH"
+  '';
+  meta = with stdenv.lib; {
+    description = "Linux devices manager for the Logitech Unifying Receiver.";
```
Comment (jagajaga) [inline pkgs/applications/misc/solaar/default.nix]: Please remove dot at the end.

File: pkgs/applications/misc/solaar/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      most devices read battery status.
+
+      It comes in two flavors, command-line and GUI. Both are able to list the
+      devices paired to a Unifying Receiver, show detailed info for each
+      device, and also pair/unpair supported devices with the receiver.
+    '';
+    license = licenses.gpl2;
+    homepage = https://pwr.github.io/Solaar/;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ];
```
Comment (jagajaga) [inline pkgs/applications/misc/solaar/default.nix]: Don't you want to be a maintainer?

File: pkgs/applications/misc/solaar/default.nix (hunk 1 above)
Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: +1,
Do we have some linter?

File: pkgs/applications/misc/solaar/default.nix (hunk 2 above)
Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: sure :)

File: pkgs/applications/misc/solaar/default.nix (hunk 1 above)
Comment (jagajaga) [inline pkgs/applications/misc/solaar/default.nix]: What is linter?

File: pkgs/applications/misc/solaar/default.nix (hunk 2 above)
Comment (jagajaga) [inline pkgs/applications/misc/solaar/default.nix]: So add yourself :)

File: pkgs/applications/misc/solaar/default.nix (hunk 1 above)
Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: something like https://lintian.debian.org/

Comment (jagajaga) [inline pkgs/applications/misc/solaar/default.nix]: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: I know there are guidelines, I was asking is there any script to validate stuff. Most of the "package" is validated because of nix language and internal structures but as it looks like there is few thinks which can be improved. Maybe we could add a check to meta field (or make it a function, I'm pretty new and not sure what would be the best)?

Comment (jagajaga) [inline pkgs/applications/misc/solaar/default.nix]: I'm working on it :)
But now there is nothing.

Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: OK, cool stuff.

**Comment by spinus** [comment]
I found two issues.
1. I tried to install package on another computer and looks like I have different set of packages there because I've got an error:
```
** (.solaar-wrapped:19916): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
/nix/store/7csfrad0pzkiqisng5lr54l8i6bdyqjl-python3.4-solaar-0.9.4/bin/..solaar-wrapped-wrapped:45: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  solaar.gtk.main()

(.solaar-wrapped:19916): GLib-GIO-CRITICAL **: g_file_new_for_path: assertion 'path != NULL' failed
solaar: error: Argument 0 does not allow None as a value
```
Probably I'm missing some dependency. I'll look into that.
2. when I use python package function the package name is changed to python3.4-solaar, but attribute name is solaar (as this is not python module, it's an app). Is it possible to build package with that python function but keep the name?

**Comment by jagajaga** [comment]
Do you build using `nix.useChroot = true;` in your `configuration.nix`?

**Comment by spinus** [comment]
@jagajaga nope, I think I had some issues when this was turned on. Should I use that?

File: pkgs/applications/misc/solaar/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ fetchurl, stdenv, makeWrapper, python3Packages , gtk3 }:
+python3Packages.buildPythonPackage rec {
+  name = "solaar-${version}";
+  version = "0.9.4";
```
Comment (wonkotron) [inline pkgs/applications/misc/solaar/default.nix]: This should be `version = "0.9.2";`

Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: o lol, that's completely correct, I'll fix that :-) thank you

**Comment by spinus** [comment]
OK, hopfuly I fixed some of the issues.

@jagajaga Btw I didn't see "testPhase", is there something like that to check if app is working after it's built?

**Comment by jagajaga** [comment]
No, the main way to test pkg is to run it yourself.

**Comment by spinus** [comment]
rebased to latest master

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -3137,6 +3137,11 @@ let
 
   socat2pre = lowPrio (callPackage ../tools/networking/socat/2.x.nix { });
 
+  solaar = callPackage ../applications/misc/solaar {};
+
+  softether_4_18 = callPackage ../servers/softether/4.18.nix { };
+  softether = softether_4_18;
```
Comment (fpletz) [inline pkgs/top-level/all-packages.nix]: This seems to be unrelated to your PR.

File: pkgs/applications/misc/solaar/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  name = "solaar-${version}";
+  src = fetchurl {
+    sha256 = "0954grz2adggfzcj4df4mpr4d7qyl7w8rb4j2s0f9ymawl92i05j";
+    url = "https://github.com/pwr/Solaar/archive/${version}.tar.gz";
+  };
+
+  buildInputs =  [
+    pkgs.gtk3
+    pkgs.python3Packages.pygobject3
+    pkgs.python3Packages.pyudev
```
Comment (fpletz) [inline pkgs/applications/misc/solaar/default.nix]: Please don't use `pkgs` here. Use `gtk3` and `python3Packages` as parameters.

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (spinus) [inline pkgs/top-level/all-packages.nix]: Oh, yea, I resolved merge conflict the wrong way, good catch, thank you.

**Comment by spinus** [comment]
@fpletz fixed.

File: pkgs/applications/misc/solaar/default.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  name = "solaar-${version}";
+  src = fetchurl {
+    sha256 = "0954grz2adggfzcj4df4mpr4d7qyl7w8rb4j2s0f9ymawl92i05j";
+    url = "https://github.com/pwr/Solaar/archive/${version}.tar.gz";
+  };
+
+  buildInputs =  [
+    gtk3
+    pygobject3
+    pyudev
```
Comment (fpletz) [inline pkgs/applications/misc/solaar/default.nix]: Travis is red. I was not very precise. The python packages are of course still in `python3Packages`. So just use `python3Packages.pygobject3` and `python3Packages.pyudev` and adjust the parameters on line 1.

Comment (spinus) [inline pkgs/applications/misc/solaar/default.nix]: So, why we want to mix packages and nix internals and we don't want to mix nix internals and packages with python packages?

**Comment by spinus** [comment]
should be fixed

**Comment by vcunat** [comment]
I wonder... has anyone even tested or used this expression? The output is just an empty directory!

**Comment by spinus** [comment]
It was working for me, :[

**Comment by vcunat** [comment]
Certainly not in the state it was pushed. Defining `installPhase` overrides calls to normal actions such as `postInstall`, for example.

**Comment by spinus** [comment]
that's very bad, I'll try to fix soon

**Comment by spinus** [comment]
@vcunat : https://github.com/NixOS/nixpkgs/pull/12685 please take a look.
